sdk/core/src/macros.rs

```diff
@@ -290,6 +290,100 @@ macro_rules! future {
     };
 }
 
+/// The following macro invocation:
+/// ```
+/// # #[macro_use] extern crate azure_core;
+/// create_request_header_cow!(ClientRequestId, CLIENT_REQUEST_ID, "builds a client request id",);
+/// ```
+/// Turns into a Header value used to construct requests.
+#[macro_export]
+macro_rules! create_request_header_cow {
+    ($name:ident, $header:ident $(, $doc:expr)?, $(($variant:ident, $value:expr)), *) => {
+        $( #[doc = $doc] )?
+        #[derive(Debug, Clone)]
+        pub struct $name(std::borrow::Cow<'static, str>);
+
+        impl $name {
+            $(
+                pub const $variant: $name = $name::from_static($value);
+            )*
+
+            pub fn new<S>(s: S) -> Self
```

Comment:
Seems like most of this stuff is shared. Can we share the definition? We can probably do all of this in one macro since the invocation is unambiguous between the two use cases.

Reply:
There needs to be some disambiguation because of either Header or AppendToUrlQuery implementation.